### DISCUSSION TITLE
can't recover code from history?

### DISCUSSION THREAD

**Discussion by hxmwr** [coder/code-server#5759] [General]
The other day, when I misdelete some block of code of a file, and I just press `Ctrl+Z` to recover my code,  for some reason I then close the online editor and when next time I open the same file and found my misdeleted code lost. I know that's because some sync failure, but is there a way like VS Code's timeline to try to recover my code from history?

**Comment by jsjoeio** [comment]
Hmm...not that I can think of 🤔 Ctrl+Z should have worked.

**Reply by hxmwr** [reply]
Maybe, I've encountered a rare case when my deleting code operation synced successfully, but later when I restore that block of code and the operation weren't synced to the server.
I think this should be taken into consideration for our precious code.

**Comment by tobycm** [comment]
You could go to `$HOME/.local/share/code-server/User/History` and see through the folders to find your old code.

**Reply by hxmwr** [reply]
<img width="520" alt="image" src="https://user-images.githubusercontent.com/9283555/204121543-d3194ab2-8b10-4345-8d97-c4d88a172bad.png"> couldn't find that folder.
